cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -371,6 +371,9 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
 
   define_option(ARROW_WITH_BACKTRACE "Build with backtrace support" ON)
 
+  define_option(ARROW_WITH_OPENTELEMETRY
+                "Build libraries with OpenTelemetry support for distributed tracing" OFF)
```

Comment:
How about adding `ARROW_TRACING` (or something) option to the "Project component" section instead?

Reply:
Hmm, perhaps. It seems a little odd since it's not its own top-level directory (and not its own user-facing API) like the other components generally are, and I think we'd want ARROW_ENGINE and perhaps ARROW_FLIGHT to require OpenTelemetry at some point.

Reply:
I see.